train_gpt.py

```diff
@@ -93,7 +158,12 @@ def step(self):
             momentum = group['momentum']
             nesterov = group['nesterov']
             ns_steps = group['ns_steps']
-            update_buffers = group['update_buffer']
+            update_buffer = group['update_buffer']
+            if "update_buffer_views" not in group:
+                group["update_buffer_views"] = [
+                    update_buffer[i].view(-1) for i in range(self.world_size)
+                ]
```

Comment:
This in `__init__` looks more compact (the codes for `update_buffer` and its views are put together)
```python
        def create_update_buffer(size: int):
            b = torch.empty(world_size, size, dtype=torch.bfloat16, device="cuda")
            return dict(update_buffer=b, update_buffer_views=[b[i] for i in range(world_size)])
        param_groups = [
            dict(params=[p for p in params if p.numel() == size], **create_update_buffer(size)) for size in sizes]
```